SiteParser.java

```diff
@@ -0,0 +1,307 @@
+package biz.redfrigate.merchantservices.web.main;
+
+import au.com.bytecode.opencsv.CSVWriter;
+import biz.redfrigate.merchantservices.web.servlet.InitParams;
+import com.amazonaws.AmazonClientException;
+import com.amazonaws.AmazonServiceException;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3Client;
+import com.amazonaws.services.s3.model.PutObjectRequest;
+import org.glassfish.jersey.servlet.ServletContainer;
+import org.jsoup.Connection;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+
+import javax.imageio.ImageIO;
+import javax.ws.rs.core.Response;
+import java.awt.image.BufferedImage;
+import java.io.*;
+import java.net.*;
+import java.nio.charset.Charset;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+import static au.com.bytecode.opencsv.CSVWriter.NO_QUOTE_CHARACTER;
+
+/**
+ * Grishko
+ * 01.04.2018
+ */
+public class SiteParser extends ServletContainer {
+
+    private final static String siteUrlForParsing = "https://service.nalog.ru/tabak.do";
+    private final static String captchaFileName = "captchaImage.jpg";
+    private static String date = ""; // example = 29.03.2018 if value "" that use current date
+    private static Connection connect;
+    public final static String resourceDirectory = InitParams.getInstance().getServletConfig().getInitParameter("resourceDirectory");
+    public final static String resultDirectory = InitParams.getInstance().getServletConfig().getInitParameter("resultDirectory");
+
+    public void parse() throws IOException {
+        deletePrevJpgAndCsvFiles();
```

Comment:
Method has name parse(), but it also modifies storage. It's confusing. It will be much better if we will have separate methods (evenmore separate classes) for such actions.